src/VarianceReduction.jl

```diff
@@ -0,0 +1,79 @@
+using Statistics:mean
+"""
+Calculate the sample variance of a given set of labels.
+
+Args:
+    y: A vector of numerical labels for which the variance is to be computed.
+
+Returns:
+    variance: The sample variance of the input label vector y.
+"""
+function variance(y::Vector{L}) where {L}
+    variance = sum((y .- mean(y)).^2) / length(y)-1
+    return variance
+end
+
+function variance_reduction(left_dataset::Vector{L}, right_dataset::Vector{L}) where {L} 
+    number_of_left_rows = length(left_dataset)  
+    number_of_right_rows = length(right_dataset)  
+    total = number_of_left_rows + number_of_right_rows  
+    variance_reduction =  ((number_of_left_rows / total) * variance(left_dataset) + (number_of_right_rows / total) * variance(right_dataset) ) 
+    return variance_reduction
+end
+
+"""
+Splits the labels into two nodes based on the provided feature and value.
+
+Args:
+    data: A matrix of features, where each row is a data point and each column is a feature.
+    labels: A vector of labels corresponding to the data points.
+    index: The index of the feature to split on.
+    value: The value to split the feature on.
+
+Returns:
+    A tuple containing the left and right sets of labels.
+  """
+function split_node_vr(data::Matrix{T}, labels::Vector{L}, index, value) where {T, L} 
+    x_index = data[:, index]
+    # if feature is numerical
+    if eltype(x_index) <: Number
```

Comment:
Here, we probably also need this for it to work:
`if eltype(identity.(x_index)) <: Number`